tmail-backend/tmail-third-party/openpaas/src/main/java/com/linagora/tmail/contact/OpenPaasContactsConsumer.java

```diff
@@ -90,6 +90,12 @@ public void start() {
         consumeContactsDisposable = doConsumeContactMessages();
     }
 
+    public void restart() {
+        Disposable previousConsumer = consumeContactsDisposable;
+        consumeContactsDisposable = doConsumeContactMessages();
+        previousConsumer.dispose();
```

Comment:
```suggestion
        Optional.ofNullable(previousConsumer).ifPresent(Disposable::dispose);
```

No it's null of not started yet.